sw/airborne/firmwares/rotorcraft/guidance/guidance_indi.c

```diff
@@ -229,10 +234,9 @@ void guidance_indi_run(bool in_flight, float heading_sp)
 
   AbiSendMsgTHRUST(THRUST_INCREMENT_ID, euler_cmd.z);
 
-  guidance_euler_cmd.phi = roll_filt.o[0] + euler_cmd.x;
-  guidance_euler_cmd.theta = pitch_filt.o[0] + euler_cmd.y;
-  //zero psi command, because a roll/pitch quat will be constructed later
-  guidance_euler_cmd.psi = 0;
+  guidance_euler_cmd.theta = pitch_filt.o[0] + euler_cmd.x;
+  guidance_euler_cmd.phi = roll_filt.o[0] + euler_cmd.y;
```

Comment:
why are `euler_cmd.x` and `euler_cmd.y` changed here?

Reply:
That is because of the rotation order, first around Y (pitch) then X (roll) and then Z (yaw). Correspondingly, I changed the input vector to be [pitch, roll, thrust]. Pitch is therefore the first element of the vector (x) and roll the second (y). It used to be [roll, pitch, thrust] for the ZYX Eulers, corresponding to [roll, pitch, yaw].